### ISSUE TITLE
CONST_INT quoted like string in ->where(['<', 'attribute', $integer]) 

### ISSUE THREAD

**Issue by spinitron** [yiisoft/yii2-sphinx#71]
### What steps will reproduce the problem?

```php
            SpinSphinx::find()
                ->select('id')
                ->match($this->q)
                ->limit(2 * $this->pageSize)
                ->where(['<', 'song_time', time()])
                ->column();
```

### What's expected?

No SQL error. The query should be 

SELECT `id` FROM `so_search_spin` WHERE (MATCH('search term')) AND (`song_time` < 1487283583) LIMIT 200

### What do you get instead?

SQLSTATE[42000]: Syntax error or access violation: 1064 sphinxql: syntax error, unexpected QUOTED_STRING, expecting CONST_INT or CONST_FLOAT or DOT_NUMBER or '-' near ''1487283583') LIMIT 200'
The SQL being executed was: SELECT `id` FROM `so_search_spin` WHERE (MATCH('search term')) AND (`song_time` < '1487283583') LIMIT 200

### Additional info

| Q                     | A
| --------------------- | ---
| Yii version           | 2.0.11.2
| Yii Sphinx version    |2.0.7
| Sphinx server version |Server version: 2.2.11-id64-release (95ae9a6)
| PHP version           |PHP 7.0.12
| Operating system      |Ubuntu 14.04

**Comment by spinitron** [comment]
Rolling back to 2.0.6 resolves the issue.

**Comment by klimov-paul** [comment]
Unable to reproduce.
Following unit test runs fine:

```php
public function testWhereCompare()
    {
        $results = ArticleIndex::find()
            ->select('id')
            ->match('about')
            ->where(['<', 'author_id', 100])
            ->limit(50)
            ->column();
        $this->assertNotEmpty($results);

        $results = ArticleIndex::find()
            ->select('id')
            ->match('about')
            ->where(['<', 'author_id', '100'])
            ->limit(50)
            ->column();
        $this->assertNotEmpty($results);
    }
```

**Comment by spinitron** [comment]
where may i find the unit test?

**Comment by spinitron** [comment]
Your test does not cover the failure case. Try instead with `'add_date'` as the compared attribute, rather than comparing ID.

**Comment by klimov-paul** [comment]
> where may i find the unit test?

You may add this code to [ActiveQueryTest](https://github.com/yiisoft/yii2-sphinx/blob/master/tests/ActiveQueryTest.php)

> Your test does not cover the failure case. Try instead with 'add_date' as the compared attribute, rather than comparing ID.

`author_id` is not an ID - it is a simple integer filed, see: https://github.com/yiisoft/yii2-sphinx/blob/master/tests/data/sphinx.conf#L26
Field naming does not change use case.

If you can compose a unit test, which reproduces your issue - post its source code here or open a PR with it.

**Comment by djagya** [comment]
@klimov-paul I think you'll get the error if you rename the field to `add_date`.
As far as I can see the problem here is that you ignore `timestamp` column type in `Schema::getColumnPhpType()`, which gives us `string` as a php column type, though Sphinx expects an integer value in the condition.

And you don't have any tests to cover that case.

**Comment by klimov-paul** [comment]
Can you provide your Sphinx config?

**Comment by tom--** [comment]
I added your test and another one for the timestamp type 

- [branched from tag 2.0.6 https://github.com/tom--/yii2-sphinx](https://github.com/tom--/yii2-sphinx/tree/71-CONST_INT_2.0.6)
- [branched from tag 2.0.7 https://github.com/tom--/yii2-sphinx](https://github.com/tom--/yii2-sphinx/tree/71-CONST_INT_2.0.7)

i used your [sphinx config](https://github.com/tom--/yii2-sphinx/blob/71-CONST_INT_2.0.7/tests/data/sphinx.conf) except changing the host to 127.0.0.1

**Comment by tom--** [comment]
Indexer output

```
 yii2-sphinx git:(71-CONST_INT_2.0.6)➜  indexer --config tests/data/sphinx.conf --all --rotate
Sphinx 2.2.11-id64-release (95ae9a6)
Copyright (c) 2001-2016, Andrew Aksyonoff
Copyright (c) 2008-2016, Sphinx Technologies Inc (http://sphinxsearch.com)

using config file 'tests/data/sphinx.conf'...
WARNING: key 'sql_query_info' was permanently removed from Sphinx configuration. Refer to documentation for details.
WARNING: key 'sql_query_info' was permanently removed from Sphinx configuration. Refer to documentation for details.
WARNING: key 'charset_type' was permanently removed from Sphinx configuration. Refer to documentation for details.
WARNING: key 'charset_type' was permanently removed from Sphinx configuration. Refer to documentation for details.
WARNING: key 'max_matches' was permanently removed from Sphinx configuration. Refer to documentation for details.
indexing index 'yii2_test_article_index'...
WARNING: attribute 'id' not found - IGNORING
collected 1002 docs, 0.1 MB
collected 5 attr values
sorted 0.0 Mvalues, 100.0% done
sorted 0.0 Mhits, 100.0% done
total 1002 docs, 88040 bytes
total 0.016 sec, 5362733 bytes/sec, 61034.29 docs/sec
indexing index 'yii2_test_item_index'...
WARNING: attribute 'id' not found - IGNORING
collected 2 docs, 0.0 MB
sorted 0.0 Mhits, 100.0% done
total 2 docs, 36 bytes
total 0.002 sec, 12354 bytes/sec, 686.34 docs/sec
indexing index 'yii2_test_item_delta_index'...
WARNING: attribute 'id' not found - IGNORING
collected 0 docs, 0.0 MB
total 0 docs, 0 bytes
total 0.002 sec, 0 bytes/sec, 0.00 docs/sec
skipping non-plain index 'yii2_test_rt_index'...
skipping non-plain index 'yii2_test_distributed'...
total 16 reads, 0.000 sec, 7.0 kb/call avg, 0.0 msec/call avg
total 33 writes, 0.000 sec, 6.0 kb/call avg, 0.0 msec/call avg
rotating indices: successfully sent SIGHUP to searchd (pid=38929).
```

**Comment by tom--** [comment]
phpunit output in 2.0.7

```
 yii2-sphinx git:(71-CONST_INT_2.0.7)➜  ../yii2/vendor/bin/phpunit
PHPUnit 4.8.23-2-g264188dd by Sebastian Bergmann and contributors.

...........E...................................................  63 / 117 ( 53%)
......................................................

Time: 2.31 seconds, Memory: 10.00Mb

There was 1 error:

1) yiiunit\extensions\sphinx\ActiveQueryTest::testWhereCompareTimestampAttr
yii\db\Exception: SQLSTATE[42000]: Syntax error or access violation: 1064 sphinxql: syntax error, unexpected QUOTED_STRING, expecting CONST_INT or CONST_FLOAT or DOT_NUMBER or '-' near ''1405000000') LIMIT 50'
The SQL being executed was: SELECT `id` FROM `yii2_test_article_index` WHERE (MATCH('about')) AND (`add_date` < '1405000000') LIMIT 50

/Volumes/Sites/yii2-sphinx/Schema.php:558
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Command.php:917
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Command.php:405
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Query.php:285
/Volumes/Sites/yii2-sphinx/tests/ActiveQueryTest.php:60
/Volumes/Sites/yii2/vendor/phpunit/phpunit/phpunit:47

Caused by
PDOException: SQLSTATE[42000]: Syntax error or access violation: 1064 sphinxql: syntax error, unexpected QUOTED_STRING, expecting CONST_INT or CONST_FLOAT or DOT_NUMBER or '-' near ''1405000000') LIMIT 50'

/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Command.php:902
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Command.php:405
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Query.php:285
/Volumes/Sites/yii2-sphinx/tests/ActiveQueryTest.php:60
/Volumes/Sites/yii2/vendor/phpunit/phpunit/phpunit:47

FAILURES!
Tests: 117, Assertions: 418, Errors: 1.
```

**Comment by tom--** [comment]
phpunit output in 2.0.6

```
 yii2-sphinx git:(71-CONST_INT_2.0.6)➜  ../yii2/vendor/bin/phpunit
PHPUnit 4.8.23-2-g264188dd by Sebastian Bergmann and contributors.

..........EE...................................................  63 / 112 ( 56%)
.................................................

Time: 2.19 seconds, Memory: 10.00Mb

There were 2 errors:

1) yiiunit\extensions\sphinx\ActiveQueryTest::testWhereCompareIntAttr
yii\db\Exception: SQLSTATE[42000]: Syntax error or access violation: 1064 sphinxql: syntax error, unexpected QUOTED_STRING, expecting CONST_INT or CONST_FLOAT or DOT_NUMBER or '-' near ''5') LIMIT 50'
The SQL being executed was: SELECT `id` FROM `yii2_test_article_index` WHERE (MATCH('about')) AND (`author_id` < '5') LIMIT 50

/Volumes/Sites/yii2-sphinx/Schema.php:556
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Command.php:917
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Command.php:405
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Query.php:285
/Volumes/Sites/yii2-sphinx/tests/ActiveQueryTest.php:49
/Volumes/Sites/yii2/vendor/phpunit/phpunit/phpunit:47

Caused by
PDOException: SQLSTATE[42000]: Syntax error or access violation: 1064 sphinxql: syntax error, unexpected QUOTED_STRING, expecting CONST_INT or CONST_FLOAT or DOT_NUMBER or '-' near ''5') LIMIT 50'

/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Command.php:902
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Command.php:405
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Query.php:285
/Volumes/Sites/yii2-sphinx/tests/ActiveQueryTest.php:49
/Volumes/Sites/yii2/vendor/phpunit/phpunit/phpunit:47

2) yiiunit\extensions\sphinx\ActiveQueryTest::testWhereCompareTimestampAttr
yii\db\Exception: SQLSTATE[42000]: Syntax error or access violation: 1064 sphinxql: syntax error, unexpected QUOTED_STRING, expecting CONST_INT or CONST_FLOAT or DOT_NUMBER or '-' near ''1405000000') LIMIT 50'
The SQL being executed was: SELECT `id` FROM `yii2_test_article_index` WHERE (MATCH('about')) AND (`add_date` < '1405000000') LIMIT 50

/Volumes/Sites/yii2-sphinx/Schema.php:556
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Command.php:917
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Command.php:405
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Query.php:285
/Volumes/Sites/yii2-sphinx/tests/ActiveQueryTest.php:68
/Volumes/Sites/yii2/vendor/phpunit/phpunit/phpunit:47

Caused by
PDOException: SQLSTATE[42000]: Syntax error or access violation: 1064 sphinxql: syntax error, unexpected QUOTED_STRING, expecting CONST_INT or CONST_FLOAT or DOT_NUMBER or '-' near ''1405000000') LIMIT 50'

/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Command.php:902
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Command.php:405
/Volumes/Sites/yii2-sphinx/vendor/yiisoft/yii2/db/Query.php:285
/Volumes/Sites/yii2-sphinx/tests/ActiveQueryTest.php:68
/Volumes/Sites/yii2/vendor/phpunit/phpunit/phpunit:47

FAILURES!
Tests: 112, Assertions: 374, Errors: 2.
```

**Comment by klimov-paul** [comment]
Resolved by commit d989eff548253b4a1f3f14414f59eb842e72dc1d
